R/sql_translate_env.R

```diff
@@ -73,15 +75,34 @@ sql_translate_env.PrestoConnection <- function(con) {
           i <- as.integer(i)
         }
         dbplyr::build_sql(x, "[", i, "]")
-      }
+      },
+      quantile = function(...) stop(quantile_error_msg(), call. = FALSE),
+      median = function(...) stop(quantile_error_msg("median"), call. = FALSE)
     ),
     sql_translator(.parent = base_agg,
       n = function() sql("COUNT(*)"),
       sd =  sql_prefix("STDDEV_SAMP"),
       var = sql_prefix("VAR_SAMP"),
       all = sql_prefix("BOOL_AND"),
-      any = sql_prefix("BOOL_OR")
+      any = sql_prefix("BOOL_OR"),
+      quantile = function(...) stop(quantile_error_msg(), call. = FALSE),
+      median = function(...) stop(quantile_error_msg("median"), call. = FALSE)
     ),
     presto_window_functions()
   ))
 }
+
+#' Create error messages for quantile-like functions
+#'
+#' @param f a string giving the name of the function
+#'
+#' @return error message for \code{f}
+#' @keywords internal
+#' @noRd
+quantile_error_msg <- function(f = "quantile") {
+  paste(
+    paste0("`", f, "()`"),
+    "is not supported in this SQL variant,",
+    "try `approx_percentile()` instead; see Presto docs"
```

Comment:
Nit: "documentation." instead of "docs".

Reply:
Done.